### PR TITLE
Race condition during startup fix 

### PR DESCRIPTION
**Pull request by andriikonstantinov**
https://github.com/cloudwan/gohan/issues/358

### CODE REVIEW

File: server/transaction_commit_informer.go [hunk 1]
```diff
@@ -25,11 +25,14 @@ import (
...
 
 	"github.com/cloudwan/gohan/schema"
+	"sync"
 )
 
 var transactionCommited chan int
-
+var mutex sync.Mutex
 func transactionCommitInformer() chan int {
+	mutex.Lock()
```
Comment (p-kozlowski) [inline server/transaction_commit_informer.go]: prefer sync.Once (cf. https://github.com/cloudwan/gohan/pull/503/files)

Comment (andriikonstantinov) [inline server/transaction_commit_informer.go]: Fixed

**Comment by andriikonstantinov** [comment]
@przemyslaw-dobrowolski-cl  please review again

File: server/transaction_commit_informer.go [hunk 2]
```diff
@@ -25,14 +25,18 @@ import (
 	"context"
 
 	"github.com/cloudwan/gohan/schema"
+	"sync"
 )
 
-var transactionCommited chan int
+var (
+	transactionCommited chan int
+	wasCommited         sync.Once
```
Comment (przemyslaw-dobrowolski-cl) [inline server/transaction_commit_informer.go]: What was commited? this name is not self-explaining. Please rename this to something more meaningful, like transactionCommitedOnce or similar

**Comment by przemyslaw-dobrowolski-cl** [comment]
@andriikonstantinov Can you provide a proof for this race condition? eg. stack trace?

**Comment by przemyslaw-dobrowolski-cl** [comment]
@andriikonstantinov Could you cite any stack trace for this face condition?

**Comment by przemyslaw-dobrowolski-cl** [comment]
@andriikonstantinov please rebase

**Comment by p-kozlowski** [comment]
@przemyslaw-dobrowolski-cl this issue hasn't been observed, it was found by code analysis: https://github.com/cloudwan/gohan/issues/358

**Comment by p-kozlowski** [comment]
please rebase

**Pull request merged**